src/__swaps__/screens/Swap/components/GestureHandlerButton.tsx

```diff
@@ -62,40 +66,59 @@ export function GestureHandlerButton({
   pointerEvents = 'box-only',
   scaleTo = 0.86,
   style,
+  tapRef,
 }: GestureHandlerButtonProps) {
-  const gesture = useMemo(
-    () =>
-      Gesture.Race(
-        Gesture.Tap()
-          .enabled(!disabled)
-          .onBegin(e => {
-            if (onPressStartWorklet) onPressStartWorklet(e);
-          })
-          .onEnd(e => {
-            if (onPressWorklet) onPressWorklet(e);
-            if (onPressJS) runOnJS(onPressJS)(e);
-          }),
-        Gesture.LongPress()
-          .enabled(!disabled && !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet))
-          .minDuration(longPressDuration)
-          .onStart(e => {
-            if (onLongPressWorklet) onLongPressWorklet(e);
-            if (onLongPressJS) runOnJS(onLongPressJS)(e);
-          })
-          .onFinalize((_, success) => {
-            if (onLongPressEndWorklet) onLongPressEndWorklet(success);
-          })
-      ),
-    [disabled, longPressDuration, onLongPressEndWorklet, onLongPressJS, onLongPressWorklet, onPressJS, onPressStartWorklet, onPressWorklet]
-  );
+  const gesture = useMemo(() => {
+    const tap = Gesture.Tap()
+      .enabled(!disabled)
+      .onBegin(e => {
+        if (onPressStartWorklet) onPressStartWorklet(e);
+      })
+      .onEnd(e => {
+        if (onPressWorklet) onPressWorklet(e);
+        if (onPressJS) runOnJS(onPressJS)(e);
+      });
+
+    if (tapRef) tap.withRef(tapRef);
+
+    const longPressEnabled = !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet);
+
+    if (!longPressEnabled) return tap;
+
+    const longPress = Gesture.LongPress()
+      .enabled(!disabled && !!(onLongPressEndWorklet || onLongPressJS || onLongPressWorklet))
+      .minDuration(longPressDuration)
+      .onStart(e => {
+        if (onLongPressWorklet) onLongPressWorklet(e);
+        if (onLongPressJS) runOnJS(onLongPressJS)(e);
+      })
+      .onFinalize((_, success) => {
+        if (onLongPressEndWorklet) onLongPressEndWorklet(success);
+      });
+
+    if (longPressRef) longPress.withRef(longPressRef);
+
+    return Gesture.Race(tap, longPress);
+  }, [
+    disabled,
+    longPressDuration,
+    longPressRef,
+    onLongPressEndWorklet,
+    onLongPressJS,
+    onLongPressWorklet,
+    onPressJS,
+    onPressStartWorklet,
+    onPressWorklet,
+    tapRef,
+  ]);
 
   return (
     <ConditionalWrap
       condition={IS_IOS}
       wrap={children => (
         <ButtonPressAnimation
           disabled={disabled}
-          minLongPressDuration={longPressDuration}
+          minLongPressDuration={longPressDuration * 1.2}
```

Comment:
Curious what the reasoning is here? Should we just change the default `LONG_PRESS_DURATION_IN_MS` to account for this? Or maybe add some ts doc to the prop to let the dev know the amount they pass in will be multiplied by `1.2`?

Reply:
Doesn't have any effect on the timing of the RNGH long press events — the buffer just ensures that on iOS, the native button press wrapper (which handles haptic feedback and the scale animation) doesn't cancel the press event before the RNGH long press events have time to fire (it cancels the press after the `minLongPressDuration` has elapsed).

If the `minLongPressDuration` is any lower than the specified `longPressDuration`, it causes the RNGH long press events to fail. Previously they were exactly equal, which was likely susceptible to race conditions.